metadata-transformer/src/main/java/no/unit/nva/metadata/filters/FilterShorterTitles.java

```diff
@@ -0,0 +1,37 @@
+package no.unit.nva.metadata.filters;
+
+import no.unit.nva.metadata.type.DcTerms;
+import org.eclipse.rdf4j.model.Literal;
+import org.eclipse.rdf4j.model.Model;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.model.Value;
+
+import java.util.Optional;
+
+/**
+ * Filters triples with property dcterms:title from the document model by length comparison, keeping the longest title.
+ */
+public final class FilterShorterTitles {
+    public static boolean apply(Model documentModel, Statement statement) {
+        return findOtherTitle(documentModel)
```

Comment:
I think this code selects randomly one of many titles and compares it with the value of the statement. If the randomly chosen title is longer than the value of the current statement the value of the statement is filtered out. 
Pay attention that there is no check whether the statement is a title. This means that an author could be filtered out by a long title.

Suggestion:

```
public final class FilterShorterTitles {

    public static boolean apply(Model documentModel, Statement statement) {
        if (isTitle(statement)) {
            String currentTitle = statement.getObject().stringValue();
            int maxTitleLength = findMaxTitleLength(documentModel);
            return currentTitle.length() < maxTitleLength;
        }
        return false;
    }

    private static int findMaxTitleLength(Model documentModel) {
        return documentModel.stream()
                   .filter(FilterShorterTitles::isTitle)
                   .map(Statement::getObject)
                   .map(Value::stringValue)
                   .map(String::length)
                   .reduce(Math::max)
                   .orElse(0);
    }

    private static boolean isTitle(Statement statement) {
        return DcTerms.TITLE.getIri().equals(statement.getPredicate())
               && statement.getObject() instanceof Literal;
    }
}
```

Reply:
The respective test is also wrong.
suggestion:
```
 @ParameterizedTest
    @CsvSource({"long, longer, longest, longest", "longest, long, longer, longest"})
    void getCreatePublicationRequestReturnsLongestTitleWhenInputContainsMultipleTitles(String first,
                                                                                       String second,
                                                                                       String third,
                                                                                       String expected)
        throws IOException {
        String shortAuthorName = createAuthorShorterThanShortestTitle(first, second, third);
        List<MetaTagPair> metaTags = List.of(new MetaTagPair(DC_TITLE, first), new MetaTagPair(CITATION_TITLE, second),

                                             new MetaTagPair(CITATION_AUTHOR, shortAuthorName));
        CreatePublicationRequest createPublicationRequest = getCreatePublicationRequest(third, metaTags);
        String actual = createPublicationRequest.getEntityDescription().getMainTitle();
        List<String> authorNames = createPublicationRequest.getEntityDescription()
                                       .getContributors()
                                       .stream()
                                       .map(Contributor::getIdentity)
                                       .map(Identity::getName)
                                       .collect(Collectors.toList());
        assertThat(authorNames, hasItem(shortAuthorName));
        assertThat(actual, equalTo(expected));
    }

    private String createAuthorShorterThanShortestTitle(String... titles) {
        return Arrays.stream(titles).reduce(this::shortestString)
                   .map(this::leftHalf)
                   .orElseThrow();
    }

    private String leftHalf(String title) {
        return title.substring(0, title.length() / 2);
    }

    private String shortestString(String left, String right) {
        return (left.length() < right.length()) ? left : right;
    }
```